### ISSUE TITLE
python-platformdirs disappeared

### ISSUE THREAD

**Issue by dhensen** [Homebrew/homebrew-core#164499]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
Error: No available formula with the name "python-platformdirs". Did you mean python-toml?
```

### Verification

- [X] My "`brew doctor` output" says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

install python-platformdirs, earlier today I ran brew doctor and I cleaned up all warnings, that led me to uninstall python-platformdirs as doctor told me to do so. I'm a brew noob. Now python needs python-platformdirs and I'm not able to install it. 

Also, this page gives 404: https://formulae.brew.sh/formula/python-platformdirs

I can't find any information about the package being deleted or deprecation warning etc.
No info on this page either: https://github.com/Homebrew/homebrew-core/pkgs/container/core%2Fpython-platformdirs

what's going on?

### What happened (include all command output)?

python-platformdirs is not available

### What did you expect to happen?

python-platformdirs will be installed

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew install python-platformdirs
```

**Comment by dhensen** [comment]
I found this: https://github.com/Homebrew/homebrew-core/commit/88078ab614b89fbd781c21b909bb3a7a3127ded5

https://github.com/Homebrew/homebrew-core/pull/163603

Why was it removed?

**Comment by dhensen** [comment]
Same for python-distlib https://github.com/Homebrew/homebrew-core/commit/6ab3c74c8c74c00e59305a4f9ded20c64d08428c

What is going on, why is stuff being deleted?

Feels like important dependencies are breaking: https://xkcd.com/2347/

**Comment by dhensen** [comment]
@cho-m Can you please help out in resolving this? thanks in advance

**Comment by dhensen** [comment]
I'm not sure if I'm drawing the right conclusion.. i'm a brew noob, coming from linux to osx. Not having the greatest experience, maybe I need to RTFM, but this is my issue after uninstalling python-platformdirs:

![image](https://github.com/Homebrew/homebrew-core/assets/2786295/52b1c1eb-ff30-4dbc-8b9a-69e5dd1d9a27)

But now I'm doubting if this is a homebrew issue, because it says it's going to use 3.9.18 to create a virtualenv (this version is pyenv managed) but then it proceeds to show 3.11 in the traceback...

**Comment by dhensen** [comment]
I'm not knowledgeable enough on brew to properly respond to this, but this situation is leaving my python stuff in a messed up situation, that can't be good. I solved all brew doctor issues. I even ran `brew list | xargs brew reinstall`, because I'm not sure how I can query which packages have old references... so lets install all was my reasoning.

Eventually I solved my pipenv issue by doing:
```
pip3.11 install platformdirs
pip3.11 install distlib
```

But I don't understand why I should be installing these by myself, while before it was done for me, by brew? Leaving this comment here for people who come searching for the same issue.
